### ISSUE TITLE
The unification commit for ideals breaks Rq / Ideal

### ISSUE THREAD

**Issue by mohamed-barakat** [oysteins/QPA2#8]
Your commit bc9e2e0 is breaking `NaturalHomomorphismByIdeal`:

```
Error, no method found! For debugging hints type ?Recovery from NoMethodFound
Error, no 2nd choice method found for `NaturalHomomorphismByIdeal' on 2 arguments at /.../gap-4.9.1/lib/methsel2.g:250 called from
NaturalHomomorphismByIdeal( A, I ) at /.../gap-4.9.1/lib/algebra.gi:1128 called from
Rq / Ideal( Rq, L ) at /.../pkg/Bialgebroids/gap/Bialgebroids.gi:450 called from
<function "unknown">( <arguments> )
 called from read-eval loop at Bialgebroid.g:26
you can 'quit;' to quit to outer loop, or
you can 'return;' to continue
brk> q;
q(u,v)[a:u->v,b:v->u]
brk> Rq;
Q * q
brk> L;
[ 1*(a*b) - 1*(u), 1*(b*a) - 1*(v) ]
brk>
```

**Comment by mohamed-barakat** [comment]
The call of
```
Rq / Ideal( Rq, L )
```
works in `Bialgebroids` and in `Bialgebroids/gap/Bialgebroids.gi:450` works with 1d94461 but breaks with bc9e2e0 with the above error message. So I guessed it could be  `NaturalHomomorphismByIdeal`.

**Comment by mohamed-barakat** [comment]
Here is an example producing the error in 1d94461:
```
q := LeftQuiver("q(u,v)[a:u->v,b:v->u]");
Rq := PathAlgebra( Rationals, q );
L := [ 1*(Rq.a*Rq.b) - 1*(Rq.u), 1*(Rq.b*Rq.a) - 1*(Rq.v) ];
Rq / Ideal( Rq, L );
```

```
Error, no method found! For debugging hints type ?Recovery from NoMethodFound
Error, no 2nd choice method found for `NaturalHomomorphismByIdeal' on 2 arguments at /.../gap-4.9.1/lib/methsel2.g:250 called from
NaturalHomomorphismByIdeal( A, I ) at /.../gap-4.9.1/lib/algebra.gi:1128 called from
<function "unknown">( <arguments> )
 called from read-eval loop at *stdin*:7
you can 'quit;' to quit to outer loop, or
you can 'return;' to continue
```

**Comment by oysteins** [comment]
When I rewrote the ideal code, I made new operations for creating ideals in quiver algebras, and happened to remove the method for `TwoSidedIdealByGenerators`. This operation is called by the builtin GAP function `Ideal`.

Thus, the `Ideal` function would now call some builtin implementation of `TwoSidedIdealByGenerators`, and we would get a "GAP" ideal instead of a "QPA" ideal.  Then when calling `Rq / I`, we invoke an implementation of `/` from GAP instead of the one in QPA.  This implementation apparently depends on `NaturalHomomorphismByIdeal`.

This is fixed in 735f26d.

Regardless of this specific problem, it still seems reasonable that we should have an implementation of `NaturalHomomorphismByIdeal`.  I have added a partial implementation (it only works for path algebras, not for quotients of path algebras) in 7ed8fc1.
